crates/ruma-signatures/src/functions.rs

```diff
@@ -234,7 +235,12 @@ pub fn verify_json(
                 _ => return Err(JsonError::not_of_type("signature", JsonType::String)),
             };
 
-            let public_key = public_keys.get(key_id).ok_or_else(|| {
+            let Ok(key_id) = OwnedKeyId::from_str(key_id) else {
```

Comment:
We can avoid to allocate here with `ServerSigningKeyId::try_from(key_id.as_str())`. And the same below.

Reply:
Not too sure how to carry that out here, since rustc complains about the size of str not being known at compilation time with:
```rust
            let Ok(key_id) = &ServerSigningKeyId::try_from(key_id.as_str()) else {
                // The key is in a format we don't support
                continue;
            };
```
both with and without the `&` here.

Reply:
Oh yeah, you need to use parentheses: `(&ServerSigningKeyId)::try_from()`